pages/_app.js

```diff
@@ -1,10 +1,36 @@
+import { useState } from "react";
 import GlobalStyle from "../styles";
+import { initialPlants } from "@/assets/plants";
 
 export default function App({ Component, pageProps }) {
+  const [plants, setPlants] = useState(initialPlants);
+  const [plantsInfo, setPlantsInfo] = useState(
+    plants.map((plant) => ({ ...plant, isMyPlant: false }))
+  );
+
+  const handleToggleMyPlants = (plant) => {
+    const updatedPlantsInfo = plantsInfo.map((info) =>
+      info.id === plant.id ? { ...info, isMyPlant: !info.isMyPlant } : info
+    );
+
+    setPlantsInfo(updatedPlantsInfo);
+  };
+
+  const isMyPlantfunction = (plant) => {
```

Comment:
Since you have the information in your 'plantsInfo' state (or in the initial data if you change it), you don't need a function to check if it is favourite or not.